contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -4,9 +4,571 @@ import "./StateChannelCommon.sol";
 import "./AStateChannelManagerProxy.sol";
 import "./StateChannelUtilLibrary.sol";
 import "./DisputeErrors.sol";
+import "./Events.sol";
 
 contract DisputeManagerFacet is StateChannelCommon {
-    /**
+    
+
+    function createDispute(
+        Dispute memory dispute,
+        bytes memory signature
+    ) public { 
+        // sanity checks
+        msg.sender == dispute.disputer;
+        // race condition check
+        if(onChainSlashes.length != onChainSlashedParticipants.length 
+        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+            revert("onChainSlashes does not match onChainSlashedParticipants");
+        }
+
+        address disputer = StateChannelUtilLibrary.retriveSignerAddress(dispute.encodedBlock, signature);
+        if(disputer != dispute.disputer) {
+            revert("Invalid signature");
+        }
+        if(proofs.length == 0) return;
+        // commit to dispute struct
+        bytes memory encodedDispute = abi.encode(dispute);
+        bytes32 memory channelIdHash = keccak256(dispute.channelId);
```

Comment:
channelId is already a hash and the unique identifier - no need to hash it further